docs/develop/base-images.md

```diff
@@ -0,0 +1,58 @@
+# Truss base images and how to work with them
+
+Base images speed up building docker images for serving as well as training from
+Trusses, by effectively building and reusing the common parts. They also help
+with container startup times when these images are used. We recently started
+using base images more extensively. This has implications for truss development.
+
+## Generating base images
+[This script](../../bin/generate_base_images.py) can be used to generate and push
+base images to dockerhub. The script allows a lot of control over which images
+to generate.
+
+```bin/generate_base_images.py --help``` to learn about all the options.
+
+Examples:
+```
+# To generate base image for live reloadable truss, with gpu support, for python
+# version 3.8, for model serving and with a version tag v0.2.2
+poetry run bin/generate_base_images.py --live-reload y --use-gpu y --python-version 3.9 --version-tag v0.2.2 --job-type server
+# Generates image named baseten/truss-server-base:3.9-gpu-reload-v0.2.2
+```
+
+```
+# Generate and pushing to dockerhub all images with version tag v0.2.2
```

Comment:
Which hub does this push to? Do folks have pull access? I image anyone running this publically will probably run into issues. Maybe we can put a disclaimer.